addons/product/models/product_template.py

```diff
@@ -178,6 +178,7 @@ def _compute_currency_id(self):
         for template in self:
             template.currency_id = template.company_id.sudo().currency_id.id or main_company.currency_id.id
 
+    @api.depends('company_id')
     @api.depends_context('company')
     def _compute_cost_currency_id(self):
         self.cost_currency_id = self.env.company.currency_id.id
```

Comment:
If it depends on the company it should probably be use in the compute also

Reply:
self.cost_currency_id = self.company_id.currency_id or self.env.company.currency_id.id